bson/registry_test.go

```diff
@@ -804,11 +804,11 @@ func TestRegistry(t *testing.T) {
 				t.Run("Decoder", func(t *testing.T) {
 					t.Parallel()
 
-					wanterr := ErrNilType
+					wanterr := "cannot perform a decoder lookup on <nil>"
 
 					gotcodec, goterr := reg.LookupDecoder(nil)
-					if !cmp.Equal(goterr, wanterr, cmp.Comparer(assert.CompareErrors)) {
-						t.Errorf("errors did not match: got %#v, want %#v", goterr, wanterr)
+					if !cmp.Equal(goterr.Error(), wanterr, cmp.Comparer(assert.CompareErrors)) {
+						t.Errorf("errors did not match: got %#v, want %q", goterr, wanterr)
```

Comment:
This should use `assert.EqualError`.

```suggestion
					assert.EqualError(t, goterr, wanterr, "errors did not match")
```